cfme/middleware/providers.py

```diff
@@ -0,0 +1,32 @@
+from cfme.fixtures import pytest_selenium as sel
+from cfme.web_ui.menu import nav
+from . import list_tbl
+
+
+nav.add_branch(
+    'middleware_providers',
+    {
+        'middleware_provider_detail':
+            lambda ctx: list_tbl.click_cells({'name': ctx['provider'].name})
+    }
+)
+
+
+class Providers(object):
```

Comment:
This is not a new provider class, but a class that will be used for navigation and validation methods for the "providers" view / sub-tab. The idea was to name these four classes as they sub-tab names "providers, "servers", "topology" and "deployments".  Perhaps a bad naming convention.